### PR TITLE
docker: add missing completions and uninstall them

### PR DESCRIPTION
**Pull request by darklore**
- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Add missing shell completion scripts installation removed in https://github.com/Homebrew/homebrew-cask/pull/124757
Fix uninstallation of shell completions.

### CODE REVIEW

**Comment by kidonng** [comment]
Hah, you beat me by a couple of minutes, I was looking at #124757 as well.

Since we are at it, could you correct those completions in `uninstall` to use `#{HOMEBREW_PREFIX}`? It was done incorrectly in #124708.

File: Casks/docker.rb [hunk 1]
```diff
@@ -33,8 +33,12 @@
   ]
 
   app "Docker.app"
+  binary "#{appdir}/Docker.app/Contents/Resources/etc/docker",
+         target: "#{HOMEBREW_PREFIX}/etc/bash_completion.d/docker"
```
Comment (bevanjkay) [inline Casks/docker.rb:37]: @darklore This is different to the previously linked bash_completion for docker that was removed in #124757 as it no longer existed. Are you intentionally linking the main `docker` binary in the bash_completion folder?

Comment (darklore) [reply Casks/docker.rb:37]: No.
This file ("#{appdir}/Docker.app/Contents/Resources/etc/docker") is not main docker binary.
This is shell completion script.

**Comment by darklore** [comment]
@kidonng Thank you for your advice.
I modified the patch to use `HOMEBREW_PREFIX` in uninstallation too.

**Review by miccal — APPROVED** [review]

**Pull request merged**